### DISCUSSION TITLE
Concept of Blocks in Peridigm

### DISCUSSION THREAD

**Discussion by jritta** [peridigm/peridigm#151] [Q&A] [answered]
Hello community,

In the Peridigm code there is a concept used which is based on so called *blocks* (defined by the class `Block`). So far, it is not completely clear to me what the *blocks* are. Do they coincide with the *blocks* specified in the input deck?
So, is a loop statement like `for(blockIt = blocks->begin() ; blockIt != blocks->end() ; blockIt++)` an iteration over the *blocks* specified by the user in the input deck?
That would also mean that the decomposition of the domain is not related to the definition of the *blocks*, right? E.g. one MPI process can have data of multiply *blocks*?

Could someone explain this in more detail?

Thank you very much in advance!
Jonas

**Comment by johntfoster** [comment, marked as answer]
Jonas,

The name "block" is something that is inherited from the Exodus database
format we use for storing and visualizing simulation results.  In Peridigm,
a block is essentially a way to specify materials that have common
materials and/or damage properties.  It has nothing to do with the parallel
domain decomposition.

Best,

John

**Reply by jritta** [reply]
Hi John,

Ok I got it. Thanks for you quick reply!
So, is it then correct to say the parallel domain decomposition functionality is done via the "map mechanism" provided by the Epetra framework of Trilinos (except the cases where wrapper of classical MPI functions is used)?

Best,
Jonas

**Comment by johntfoster** [comment]
Yes.
